please/build.go

```diff
@@ -17,31 +17,46 @@ func Build(plz, target string) ([]string, error) {
 
 // RecursivelyProvide queries the target, checking if it provides a different target for the given requirement, and
 // returns that, repeating the operation if that target also provides a different target.
-func RecursivelyProvide(plz, target, requires string) (string, error) {
+func RecursivelyProvide(plz string, target string, requires string) ([]string, error) {
 	out, err := execPlease(plz, "query", "print", "--json", "--field=provides", target)
 	if err != nil {
-		return "", err
+		return []string{}, err
```

Comment:
nit: I would prefer `nil` here to avoid allocations and make the code more in line with the rest of the project

Reply:
Done